src/lib/services/auth/index.ts

```diff
@@ -26,14 +25,39 @@ export function getTokenEndpoint() {
 export * as identities from './service/identities.js';
 export * as oauth2 from './service/oauth2/index.js';
 
-export type Token = ITokenResponse & {
+export type Token = {
+  access_token: string;
+  scope: string;
+  expires_in: number;
+  token_type: string;
   resource_server: string;
-  id_token?: string;
+  refresh_token?: string;
+};
+
+export type TokenWithRefresh = Token & {
+  refresh_token: string;
 };
+
 /**
  * @see https://docs.globus.org/api/auth/reference/#authorization_code_grant_preferred
  */
 export type TokenResponse = Token & {
   state: string;
-  other_tokens: Token[];
+  id_token?: string;
+  other_tokens?: Token[];
 };
+
+export function isToken(check: unknown): check is Token {
+  return typeof check === 'object' && check !== null && 'access_token' in check;
```

Comment:
Any array will pass the first two conditions here

Reply:
I _think_ that is ok, since the last check is a bit more explicit in ensuring there is a `access_token` property.

I am debating how far this should go, though...

Reply:
Yeah that's true, it does seem unlikely that an Array would have that property set on it